### PR TITLE
feat: implemented personal builder profile page

### PR DESCRIPTION
**Pull request by michojekunle**
## Description
Implemented personal builder profile page, that shows my details, my avatar, a short bio and my address

## Additional Information
- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

Your ENS/address: 0x7429CbD5eD20736645723E972bE60B7F6BF5959c

image

### CODE REVIEW

**Comment by michojekunle** [comment]
@phipsae @technophile-04 made this PR as regards this issue https://github.com/BuidlGuidl/batch10.buidlguidl.com/issues/9

**please review**

File: package-lock.json
Comment (phipsae) [inline package-lock.json:1]: We don't want the package-lock.json file, as we use yarn. So please remove this file.

File: packages/nextjs/public/assets/dvm0x742_avatar.jpg
Comment (phipsae) [inline packages/nextjs/public/assets/dvm0x742_avatar.jpg:1]: Here you could also name the folder after your name, so you have a dedicated folder for your page.

File: packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
+"use client";
+
+import { useCallback, useEffect, useState } from "react";
+import Image from "next/image";
+import Link from "next/link";
+import { Check, Code, Copy, Cpu, Github, Globe, Linkedin, Twitter } from "lucide-react";
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:6]: For Github, LinkedIn, Twitter it gives me deprecated warning. Maybe you want to consider to change to another library, like react icons.

File: yarn.lock
Comment (phipsae) [inline yarn.lock:1]: The only code change should be the addition of "lucide-react."

File: packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                <p className="text-secondary-content flex flex-col gap-1">
+                  <span>Frontend Developer / Smart Contract Writer</span>
+                  <span className="flex items-center text-sm opacity-70">
+                    0x7429CbD5eD20736645723E972bE60B7F6BF5959c
+                    {copied ? (
+                      <Check className="w-3 h-3 ml-2 text-green-500 transition" />
+                    ) : (
+                      <Copy className="w-3 h-3 ml-2 cursor-pointer transition" onClick={handleCopy} />
+                    )}
+                  </span>
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:123]: You might consider using the SE2 component Address to leverage the existing component. There you already have the copy functionality.

File: packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  }, [handleScroll, handleMouseMove]);
+
+  const parallaxLayers = [
+    { icon: <Code size={48} />, speed: 0.05, initialPos: { x: "18%", y: "14%" } },
+    { icon: <Cpu size={48} />, speed: 0.1, initialPos: { x: "80%", y: "50%" } },
+    { icon: <Globe size={48} />, speed: 0.15, initialPos: { x: "34%", y: "78%" } },
+  ];
+
+  return (
+    <div className="relative min-h-screen overflow-hidden bg-gradient-to-br from-base-200 to-base-300">
```
Comment (phipsae) [inline packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:59]: Your Floating Particles work perfectly in light mode. 
But in dark mode you cannot see it, so you might adapt that.

File: packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx
Comment (phipsae) [inline packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:1]: In my console when I load your page I get the following error, please fix it. 
It has to do with Parallax Background and Floating Particles.
Screenshot 2024-10-18 at 13 47 11

**Review by phipsae — CHANGES_REQUESTED** [review]
@michojekunle thanks for the page! You really like that you integrate the particles effect :)

Below you'll find comments regarding the code. If there are any questions feel free to ask. 

Thanks!

**Comment by michojekunle** [comment]
Hi @phipsae Thank you so much for your review, I've made the necessary and suggested changes as specified in the comments, and I also learnt new stuff regarding disabling SSR for client pages.

Please review 🙌🏾

> @michojekunle thanks for the page! You really like that you integrate the particles effect :)
> 
> Below you'll find comments regarding the code. If there are any questions feel free to ask.
> 
> Thanks!

Hi @phipsae Thank you so much for your review, I've made the necessary and suggested changes as specified in the comments, and I also learnt new stuff regarding disabling SSR for client pages.

Please review 🙌🏾

**Review by phipsae — CHANGES_REQUESTED** [review]
Works and great! 

Just a little change and then we are good to go merging. 

Thanks!

File: packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import dynamic from "next/dynamic";
+
+const DevvMichaelProfile = dynamic(() => import("./_components/DevvMichaelProfile"), { ssr: false });
+
+export default DevvMichaelProfile;
```
[the remark this answers is not in the document]
Comment (michojekunle) [reply packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:5]: Alright, Awesome, I'm on it

[the remark this answers is not in the document]
Comment (michojekunle) [reply packages/nextjs/app/builders/0x7429CbD5eD20736645723E972bE60B7F6BF5959c/page.tsx:6]: Hi @phipsae, Thank you so much for the review. I've implemented the changes

**Comment by phipsae** [comment]
Very nice, thank you a lot!!! 
Gonna merge it!

**Pull request merged**